tests/scipy/solve_triangular.py

```diff
@@ -0,0 +1,26 @@
+import math
+
+try:
```

Comment:
I would combine the two imports: try to import everything from `ulab`, and then, on exception, from `numpy`, and `scipy`.